### PR TITLE
make AmiciObjective only report whats being asked for

### LINKED ISSUE #776: AggregateObjective evaluation fails for `MODE_RES` and `sensi_orders=1`
AmiciObjective returns CHI2, RES, SRES, but prior (correctly) only returns SRES, leading to failure when aggregating.

### PR DESCRIPTION
**Pull request by FFroehlich**
fixes #776

### CODE REVIEW

File: pypesto/objective/amici_util.py [hunk 1]
```diff
@@ -286,7 +286,7 @@ def get_error_output(
         amici_model: AmiciModel,
         edatas: Sequence['amici.ExpData'],
         rdatas: Sequence['amici.ReturnData'],
-        sensi_order: int,
+        sensi_orders: Tuple[int],
```
Comment (yannikschaelte) [inline pypesto/objective/amici_util.py:289]: I think `Tuple[int, ...]` would be more correct https://docs.python.org/3/library/typing.html#typing.Tuple

**Review by yannikschaelte — APPROVED** [review]

File: pypesto/objective/amici_calculator.py [hunk 2]
```diff
@@ -71,6 +71,11 @@ def __call__(self,
             requested).
         """
         # set order in solver
+        if sensi_orders:
+            sensi_order = max(sensi_orders)
+        else:
+            sensi_order = 0
+
```
Comment (PaulJonasJost) [inline pypesto/objective/amici_calculator.py:78]: If sensi_order is only used in the amici_solver.setSensitivityOrder, would perhaps do it like in the other changes with 
```python
if 2 in sense_orders:
...
```

File: pypesto/objective/amici_calculator.py [hunk 3]
```diff
@@ -108,14 +113,14 @@ def __call__(self,
...
-            rdatas=rdatas, sensi_order=sensi_order, mode=mode,
+            rdatas=rdatas, sensi_orders=sensi_orders, mode=mode,
             amici_model=amici_model, amici_solver=amici_solver, edatas=edatas,
             x_ids=x_ids, parameter_mapping=parameter_mapping,
             fim_for_hess=fim_for_hess)
 
 
 def calculate_function_values(rdatas,
-                              sensi_order: int,
+                              sensi_orders: Tuple[int],
```
Comment (PaulJonasJost) [inline pypesto/objective/amici_calculator.py:123]: Here then just as yannik suggested also `Tuple[int, ...]`?

File: pypesto/objective/amici_util.py [hunk 4]
```diff
@@ -320,7 +320,10 @@ def get_error_output(
     return filter_return_dict(ret)
 
 
-def init_return_values(sensi_order, mode, dim, error=False):
+def init_return_values(sensi_orders: Tuple[int],
```
Comment (PaulJonasJost) [inline pypesto/objective/amici_util.py:323]: Here `Tuple[int, ...]`as well then.

**Review by PaulJonasJost — APPROVED** [review]
I am not too sure how convenient it is, having to pass on all wanted sensitivities together. Is there a case where I would want second order sensitivities but not first order?

File: pypesto/objective/amici_calculator.py (hunk 2 above)
Comment (FFroehlich) [reply pypesto/objective/amici_calculator.py:78]: Amici will always only compute lower sensitivity orders, as they come more or less for free. For other objectives that might not be the case, hence the distinction.

**Comment by FFroehlich** [comment]
> I am not too sure how convenient it is, having to pass on all wanted sensitivities together. Is there a case where I would want second order sensitivities but not first order?

For AmiciObjective I agree it does not make too much sense, but that's the way we implemented the API for general objectives and thats how, e.g., priors behave.

**Pull request merged**